docs/personal-designs.rst

```diff
@@ -0,0 +1,148 @@
+Compiling and Running Your Own Designs Using the Symbiflow-examples Toolchain 
+==============================================================================
+
+This section describes how to compile and download your own designs to an FPGA using only the symbiflow-examples tool chain.
+
+Setup Prior to Running the Toolchain
+--------------------------------------
+Before building any examples, you will need to first install the toolchain. To do this, follow the steps in `Getting Symbiflow <getting-symbiflow.html>`_.
+After you have downloaded the toolchain, set the installation directory to match what you set it to earlier, for example:
+
+.. code-block:: bash
+   :name: export-install-dir
+
+   export INSTALL_DIR=~/opt/symbiflow
+
+Select your FPGA family:
+
+.. tabs::
+
+   .. group-tab:: Artix-7
+
+      .. code-block:: bash
+         :name: fpga-fam-xc7
+
+         FPGA_FAM="xc7"
+
+   .. group-tab:: EOS S3
+
+      .. code-block:: bash
+         :name: fpga-fam-eos-s3
+
+         FPGA_FAM="eos-s3"
+
+Next, set the path and source for your conda environment:
+
+.. code-block:: bash
+   :name: conda-prep-env
+
+   export PATH="$INSTALL_DIR/$FPGA_FAM/install/bin:$PATH";
+   source "$INSTALL_DIR/$FPGA_FAM/conda/etc/profile.d/conda.sh"
+
+Finally, activate your Conda environment:
+
+.. code-block:: bash
+   :name: conda-act-env
+
+   conda activate $FPGA_FAM
+
+
+.. note::
+
+   You will need to run the commands for setting the path and source of your conda environment each time you open a new terminal. You will also need to activate the Conda environment for your hardware before you attempt to build your designs. It might be a good idea to add the above commands to your ``.bashrc`` either as default commands that run each time you open a new terminal or aliases to save yourself some repetitive typing. 
+
+
+Preparing Your Design 
+----------------------
+Building a design in symbiflow requires three parts, the HDL files for your design, a constraints file, and a Makefile. For simplicity, all three of these design files should be moved to a single directory. The location of the directory does not mater as long as the three design files are all within it.
+
+HDL Files
+++++++++++
+Symbiflow provides support for both Verilog and SystemVerilog HDL code. Use whichever method you prefer and add your design files to the directory of choice. If you are using the provided Makefiles to build your design, the top level module in your HDL code should be declared as ``module top (...``. Failure to do so will result in an error during the build process stating something similar to ``ERROR: Module 'top' not found!``
+
+
+
+Constraint File
+++++++++++++++++
+The Symbiflow tool chain supports both .XDC and .PCF+.SDC formats for constraints. Use whichever method you prefer and add your constraint file(s) to your design directory.
```

Comment:
A description of what XDC, PCF and SDC file should end up in the former `understanding-makefiles` document.

Reply:
I believe I have that already in customizing-makefiles->Constraint files. Are you saying I should explain how the XDC, PCF and SDC files should be written or something else? Could you clarify?

Reply:
Yeah, I meant to add some notes on what are those constraint files and what content they should have. For instance, the XDC defines can define both IOPAD constraints, IOSETTINGS and also clock constraints.

A PCF file can define the constraints for IOPADs only, while SDC is used for clock constraints needed for VTR.

If you, for instance, set only an XDC constraint file, containing both IOPAD constraints and clock constraints, an SDC will be automatically generated to provide clock constraints to VTR. Otherwise, you need to provide the SDC explicitly.

So there are two main ways of adding constraints:
1. PCF (iopad) +SDC (clock)
2. XDC (both iopads and clock constraints)